### ISSUE TITLE
Option to stay true to source color

### ISSUE THREAD

**Issue by JimVanEeden** [pchmn/expo-material3-theme#15]
Is there a plan to support the following option as can be seen at https://material-foundation.github.io/material-theme-builder/?

<img width="547" alt="image" src="https://github.com/user-attachments/assets/2b03d71c-d0f7-4ea9-95e5-f6532a257b9a">

For my app it would be great to supply a theme color and have that being used as primary color as well. Apparently the color library does support that as it's an option in the theme builder. For now I'm overriding the primary color manually, but as can be seen, some other colors change as well when you enable the option.

Support for this option would be very much appreciated! Thanks

**Comment by pchmn** [comment]
Yes it seems to be called "color fidelity" (https://m3.material.io/styles/color/advanced/adjust-existing-colors#cb49eeb4-3bbd-4521-9612-0856c27f91ef), and `@material/material-color-utilities` supports it in last versions apparently. 

I'll look into it !

**Comment by pchmn** [comment]
Sorry for the delay, but you can try beta version [`@pchmn/expo-material3-theme@1.4.0-beta.1`](https://www.npmjs.com/package/@pchmn/expo-material3-theme/v/1.4.0-beta.1) with color fidelity option:
```ts
const { theme, updateTheme } = useMaterial3Theme({ sourceColor: '#3E8260', colorFidelity: true });
// ...
updateTheme('#3E8260', { colorFidelity: true });
```

Tell me if it works for you, and I'll published it under stable version !

**Comment by pchmn** [comment]
@JimVanEeden did you have time to test it ?

**Comment by JimVanEeden** [comment]
Thanks a lot for the effort! I'm excited to try it out, however I don't think I'll be touching the app for a couple of weeks, so I will post the update later. Of course if anybody else tries it out in the meantime, please let us know :)

**Comment by Neyxo** [comment]
I just tried it out my first impression was: works great!

However:
The reason I wanted to try it is, that I noticed that for some colors (especially vibrant ones like orange), it does not match the source color at all and thought the color fidelity option might solve that. 

While the option makes the primaryContainer close to the source, upon closer testing I noticed two things:
- It makes primary and secondary colors super close together, kinda defeats having a secondary at all
- The tertiary color is now completely different and in some cases a bit agressive

But if I understood it correctly the colors come mostly from `@material/material-color-utilities`?
Also I'm not sure if this is actually working as intended and I'm just misunderstanding/disagreeing with the Material design color standard.

Here some examples of the same color with / without the color fidelity setting:

<img width="372" alt="Image" src="https://github.com/user-attachments/assets/4e695269-71d1-4536-9af9-249ae2029e92" />

<img width="374" alt="Image" src="https://github.com/user-attachments/assets/1172af8f-4dc3-4958-a265-465720f1381c" />

<img width="373" alt="Image" src="https://github.com/user-attachments/assets/b9115d30-32cf-4277-a14e-b959f71366d1" />

<img width="369" alt="Image" src="https://github.com/user-attachments/assets/0b57b89c-696c-43cb-b25d-a3b682164779" />

<img width="369" alt="Image" src="https://github.com/user-attachments/assets/83735bff-2553-4e09-b473-620357d2e27d" />

<img width="369" alt="Image" src="https://github.com/user-attachments/assets/1226bc25-2e91-477e-83d6-c83955261053" />

**Comment by Neyxo** [comment]
So I've tried to debug this a bit, since the colors also seem to mismatch the one from the `material-theme-builder`. Unfortunately their code seems to be closed source.
I guess its either an upstream issue in the typescript variant of `@material/material-color-utilities` (`material-theme-builder` is apparently written in Dart). Or, more likely, `material-theme-builder` does some custom magic.
I for myself will probably continue to use the normal mode and just override the primaryContainer color manually, since thats close enough to what I need.

**Comment by pchmn** [comment]
Yes the colors come from the official `@material/material-color-utilities` and it's supposed to follow material design guidelines.

It seems there are difference between dart and typescript libraries of `material-color-utilities` : https://github.com/material-foundation/material-color-utilities/issues/162.
This could explain the differences with https://material-foundation.github.io/material-theme-builder
